### PR TITLE
comments from merge to main

### PR DESCRIPTION
**Pull request by juani-lastra**
- [x] Eliminar el override fun onBackPressed() del form

- [x] please change the name "Arroba" to "AT"
 
- [x] What happens if location comes with CoroutineResult.Failure?
 
- [x] Styles linea 55: do not hardcode pls, use dimens
 
- [x] Styles lines 125, 63,51,52,11,12,119,133: Do not put hints in styles as they are specific for each field. Add the hint in the view and remove this style
 
- [x] Search picker linea 98: please use representative names for this let{}

Test Coverage

image

### CODE REVIEW

File: app/src/test/java/com/mobiletandil/stockapp/viewmodel/AssetFormViewModelTest.kt [hunk 1]
```diff
@@ -151,10 +151,17 @@ class AssetFormViewModelTest {
...
+        )
+
+        Assert.assertEquals(
+            AssetFormViewModel.AssetFormState.INIT_UI_WITH_BARCODE,
+            liveData.observedValues[1]?.peekContent()?.state
         )
+
         Assert.assertEquals(
-            TEST_BARCODE, liveData.observedValues.first()?.peekContent()?.barcode
+            TEST_BARCODE, liveData.observedValues[1]?.peekContent()?.barcode
```
Comment (mauroLonghi) [inline app/src/test/java/com/mobiletandil/stockapp/viewmodel/AssetFormViewModelTest.kt:164]: you have here and bellow severals hardcodes , pls remove

Comment (juani-lastra) [reply app/src/test/java/com/mobiletandil/stockapp/viewmodel/AssetFormViewModelTest.kt]: Done

**Review by mauroLonghi — APPROVED** [review]

File: app/src/main/java/com/mobiletandil/stockapp/activity/AssetFormActivity.kt [hunk 2]
```diff
@@ -81,6 +81,13 @@ class AssetFormActivity : AppCompatActivity() {
             }
             AssetFormState.FIREBASE_FAILED -> {
                 loader.dismiss()
+                showSimpleAlertDialog(
+                    this,
+                    getString(R.string.fragment_no_db_connection_description),
+                    getString(R.string.asset_form_dialog_button_ok_title),
+                    { startActivity(Intent(this, MainActivity::class.java)) },
```
Comment (SartiMau) [inline app/src/main/java/com/mobiletandil/stockapp/activity/AssetFormActivity.kt:88]: this should calll the viewmodel, there set an state and the activity react to that event with this startActivity

File: app/src/main/java/com/mobiletandil/stockapp/activity/ScannerActivity.kt [hunk 3]
```diff
@@ -149,6 +149,16 @@ class ScannerActivity : AppCompatActivity() {
                 ScanBarcodeStatus.LOADING -> {
                     loader.show()
                 }
+                ScanBarcodeStatus.FIREBASE_FAILED -> {
+                    loader.dismiss()
+                    showSimpleAlertDialog(
+                        this,
+                        getString(R.string.fragment_no_db_connection_description),
+                        getString(R.string.asset_form_dialog_button_ok_title),
+                        { startActivity(Intent(this, MainActivity::class.java)) },
```
Comment (SartiMau) [inline app/src/main/java/com/mobiletandil/stockapp/activity/ScannerActivity.kt:158]: this should calll the viewmodel, there set an state and the activity react to that event with this startActivity

File: app/src/main/java/com/mobiletandil/stockapp/fragment/SearchPickerFragment.kt [hunk 4]
```diff
@@ -97,25 +98,32 @@ class SearchPickerFragment : DialogFragment() {
...
                 this.dismiss()
             }
+            SearchPickerStatus.FIREBASE_FAILED -> {
+                loader.dismiss()
+                activity?.let {
+                    showSimpleAlertDialog(
+                        it,
+                        getString(R.string.fragment_no_db_connection_description),
+                        getString(R.string.asset_form_dialog_button_ok_title),
+                        { startActivity(Intent(requireContext(), MainActivity::class.java)) },
```
Comment (SartiMau) [inline app/src/main/java/com/mobiletandil/stockapp/fragment/SearchPickerFragment.kt:118]: this should calll the viewmodel, there set an state and the fragment react to that event with this startActivity

File: app/src/test/java/com/mobiletandil/stockapp/viewmodel/AssetFormViewModelTest.kt [hunk 5]
```diff
@@ -216,13 +256,13 @@ class AssetFormViewModelTest {
         runBlocking { assetFormViewModel.saveAsset(TEST_BARCODE, ASSET_NAME, MAIL_VALID, LOCATION).join() }
 
         Assert.assertEquals(
-            successSaveList[0].state,
-            liveData.observedValues[0]?.peekContent()?.state
+            successSaveList[FIRST_VALUE].state,
```
Comment (SartiMau) [inline app/src/test/java/com/mobiletandil/stockapp/viewmodel/AssetFormViewModelTest.kt:259]: I think the the constants FIRST_VALUE and SECOND_VALUE are not necessary, to grants readability

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/test/java/com/mobiletandil/stockapp/viewmodel/AssetFormViewModelTest.kt (hunk 5 above)
Comment (juani-lastra) [reply app/src/test/java/com/mobiletandil/stockapp/viewmodel/AssetFormViewModelTest.kt:259]: @mauroLonghi request that

File: app/src/main/java/com/mobiletandil/stockapp/activity/AssetFormActivity.kt (hunk 2 above)
Comment (juani-lastra) [reply app/src/main/java/com/mobiletandil/stockapp/activity/AssetFormActivity.kt]: Done

File: app/src/main/java/com/mobiletandil/stockapp/activity/ScannerActivity.kt (hunk 3 above)
Comment (juani-lastra) [reply app/src/main/java/com/mobiletandil/stockapp/activity/ScannerActivity.kt]: Done

File: app/src/main/java/com/mobiletandil/stockapp/fragment/SearchPickerFragment.kt (hunk 4 above)
Comment (juani-lastra) [reply app/src/main/java/com/mobiletandil/stockapp/fragment/SearchPickerFragment.kt]: Done

**Review by SartiMau — APPROVED** [review]

File: app/src/main/java/com/mobiletandil/stockapp/activity/AssetFormActivity.kt [hunk 6]
```diff
@@ -159,6 +159,7 @@ class AssetFormActivity : AppCompatActivity() {
             AssetFormState.ASSET_BY_BARCODE_DO_NOT_EXIST -> {
                 enableFieldsButBarcode()
             }
+            AssetFormState.GO_TO_MAIN_ACTIVITY -> startActivity(Intent(this, MainActivity::class.java))
```
Comment (mauroLonghi) [inline app/src/main/java/com/mobiletandil/stockapp/activity/AssetFormActivity.kt:162]: the same comment as mau, you need to call assetViewModel.goToMainActivity() here, its the same case

Comment (juani-lastra) [reply app/src/main/java/com/mobiletandil/stockapp/activity/AssetFormActivity.kt:162]: this is the state that comes from viewmodel

**Pull request merged**